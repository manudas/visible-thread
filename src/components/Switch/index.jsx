import React from 'react';
import PropTypes from 'prop-types';

import './styles.scss';

const Switch = ({
    name,
    checked,
    onChange,
    optionLabels,
}) => {
    return (
        <div className="toggle-switch m-5"
            data-testid="toggle-switch"
            onClick={e => onChange(e)}
        >
            <input
                type="checkbox"
                className="toggle-switch-checkbox"
                name={name}
                id={name}
                checked={checked}
                readOnly
                onClick={(e) => e.stopPropagation()}
            />
            <label className="toggle-switch-label" htmlFor={name}>
                <span className="toggle-switch-inner"
                    data-yes={optionLabels[0]}
                    data-no={optionLabels[1]}
                />
                <span className="toggle-switch-switch" />
            </label>
        </div>
    );
}

Switch.propTypes = {
    checked: PropTypes.bool.isRequired,
    name: PropTypes.string,
    onChange: PropTypes.func,
    optionLabels: PropTypes.arrayOf(PropTypes.string),
}

Switch.defaultProps = {
    name: "switch",
    onChange: Function.prototype,
    optionLabels: ["Yes", "No"],
}

export default Switch;
