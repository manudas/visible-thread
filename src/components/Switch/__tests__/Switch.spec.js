import React from 'react';
import {
    fireEvent,
    render,
} from '@testing-library/react';
import Switch from '..';

describe('Switch test suite', () => {

    let handler;
    let renderer;

    beforeEach(() => {
        handler = jest.fn();
        renderer = render(
            <Switch
                checked
                onChange={handler}
            />
        );
    });

    it('should render <Switch />', () => {
        const { container } = renderer;
        expect(container).toMatchSnapshot();
    });

    it('should call onChange function', async () => {

        const { getByTestId } = renderer;
        const container = getByTestId('toggle-switch');
        fireEvent.click(container);

        expect(handler).toHaveBeenCalledTimes(1);
    });
});
