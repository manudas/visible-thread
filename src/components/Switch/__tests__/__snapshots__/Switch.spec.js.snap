// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch test suite should render <Switch /> 1`] = `
<div>
  <div
    class="toggle-switch m-5"
    data-testid="toggle-switch"
  >
    <input
      checked=""
      class="toggle-switch-checkbox"
      id="switch"
      name="switch"
      readonly=""
      type="checkbox"
    />
    <label
      class="toggle-switch-label"
      for="switch"
    >
      <span
        class="toggle-switch-inner"
        data-no="No"
        data-yes="Yes"
      />
      <span
        class="toggle-switch-switch"
      />
    </label>
  </div>
</div>
`;
